src/coord/src/coord.rs

```diff
@@ -87,26 +87,32 @@ mod dataflow_builder;
 pub enum Message {
     Command(Command),
     Worker(WorkerFeedbackWithMeta),
-    AdvanceSourceTimestamp {
-        id: SourceInstanceId,
-        update: TimestampSourceUpdate,
-    },
-    StatementReady {
-        session: Session,
-        tx: ClientTransmitter<ExecuteResponse>,
-        result: Result<sql::ast::Statement, anyhow::Error>,
-        params: Params,
-    },
-    SinkConnectorReady {
-        session: Session,
-        tx: ClientTransmitter<ExecuteResponse>,
-        id: GlobalId,
-        oid: u32,
-        result: Result<SinkConnector, anyhow::Error>,
-    },
+    AdvanceSourceTimestamp(AdvanceSourceTimestamp),
+    StatementReady(StatementReady),
+    SinkConnectorReady(SinkConnectorReady),
     Shutdown,
 }
 
+pub struct AdvanceSourceTimestamp {
+    pub id: SourceInstanceId,
+    pub update: TimestampSourceUpdate,
+}
+
+pub struct StatementReady {
+    pub session: Session,
+    pub tx: ClientTransmitter<ExecuteResponse>,
+    pub result: Result<sql::ast::Statement, anyhow::Error>,
+    pub params: Params,
+}
+
+pub struct SinkConnectorReady {
+    pub session: Session,
+    pub tx: ClientTransmitter<ExecuteResponse>,
+    pub id: GlobalId,
+    pub oid: u32,
+    pub result: Result<SinkConnector, anyhow::Error>,
+}
```

Comment:
As annoying as this is, I do think it's exactly the right approach! I've come to really dislike all but the simplest struct enum variants because you almost always wind up splitting them out like this eventually.